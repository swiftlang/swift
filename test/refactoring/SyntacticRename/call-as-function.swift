struct Adder {
    var base: Int
    func /*test:def*/callAsFunction(x: Int, y: Int) -> Adder {
        return self
    }
}

let add3 = Adder(base: 3)
_ = add3/*test:call*/(x: 10, y: 11)
_ = add3 . /*test:call*/callAsFunction(x: 10, y: 11)
_ = add3 . /*test:ref*/callAsFunction(x:y:)
_ = add3 . /*test:ref*/callAsFunction
_ = (add3 . /*test:call*/callAsFunction())/*test:call*/(x: 10, y: 11)

// REQUIRES: swift_swift_parser
// RUN: %empty-directory(%t.ranges)
// RUN: %refactor -find-rename-ranges -source-filename %s -pos="test" -is-function-like -old-name "callAsFunction(x:y:)" >> %t.ranges/call-as-function.swift.expected
// RUN: diff -u %S/Outputs/call-as-function.swift.expected %t.ranges/call-as-function.swift.expected
