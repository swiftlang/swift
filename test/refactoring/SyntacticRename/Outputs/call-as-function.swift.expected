struct Adder {
    var base: Int
    func /*test:def*/<keywordBase>callAsFunction</keywordBase>(<arglabel index=0>x</arglabel><param index=0></param>: Int, <arglabel index=1>y</arglabel><param index=1></param>: Int) -> Adder {
        return self
    }
}

let add3 = Adder(base: 3)
_ = add3/*test:call*/(<callarg index=0>x</callarg><callcolon index=0>: </callcolon>10, <callarg index=1>y</callarg><callcolon index=1>: </callcolon>11)
_ = add3 . /*test:call*/<keywordBase>callAsFunction</keywordBase>(<callarg index=0>x</callarg><callcolon index=0>: </callcolon>10, <callarg index=1>y</callarg><callcolon index=1>: </callcolon>11)
_ = add3 . /*test:ref*/<keywordBase>callAsFunction</keywordBase>(<sel index=0>x</sel>:<sel index=1>y</sel>:)
_ = add3 . /*test:ref*/<keywordBase>callAsFunction</keywordBase>
_ = (add3 . /*test:call*/<keywordBase>callAsFunction</keywordBase>())/*test:call*/(<callarg index=0>x</callarg><callcolon index=0>: </callcolon>10, <callarg index=1>y</callarg><callcolon index=1>: </callcolon>11)

